src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -855,10 +884,23 @@ private static string FormatDiagnostics(ImmutableArray<DiagnosticAnalyzer> analy
                 }
                 else
                 {
-                    AppendLocation(diagnostics[i].Location);
-                    foreach (var additionalLocation in diagnostics[i].AdditionalLocations)
+                    // The unnecessary code designator is ignored for the primary diagnostic location.
+                    AppendLocation(diagnostics[i].Location, isUnnecessary: false);
```

Comment:
>diagnostics[i] [](http://example.com/codeflow?start=35&length=14)

nit: feels like this code would be a bit simpler if diagnostics[i] was just in a local

Reply:
➡️ Simplified